### PR TITLE
[Service Bus Client] Review and Clean Local Assets (#5354)

### PR DESCRIPTION
**Pull request by jsquire**
## Summary

The goal of this set of changes is intended to tidy some of the local assets associated with the Service Bus client library, which were absorbed when the source code was adopted into the central repository.    This scope of this work is intended to be a focused clean-up, with the code and other artifacts otherwise left as they were in the [standalone repository](https://github.com/Azure/azure-service-bus-dotnet), including the existing code formatting and conventions.

## Areas  for Discussion

**UPDATE:** per the discussion in the Teams process channel, consensus was reached that the file was not needed.  It has been removed from the repository and this pull request.

  An authoritative opinion on the `thirdpartynotices.txt` file that has been moved to the repository root would be appreciated.  This notice existed within the Service Bus stand-alone repository;  I am not sure whether or not it is needed and relevant here or could be safely deleted.   It was suggested in a Teams thread that @kurtzeborn and @mikeharder may be good individuals to weigh in on this.

### Repository Structure
  - Moving the third party notifications from the Serivce Bus SDK into the root and generalizing.

### Service Bus Client 
  - Updating the API tests to properly clean up their temporary files on failure.  

  - Removing the local .gitignore, as there is no longer any content specific to the Service Bus library.

  - Restoring the original content of the local ReadMe, which was inadvertently destroyed during a rebase.  (**Note:** this intentionally leaves the content in a non-compliant state with this repository's norms.  That will be addressed in a [future set of changes.](https://github.com/Azure/azure-sdk-for-net/issues/5349))

  - Minor tweaks to the local ReadMe for asset location updates.  

  - Restoring the test infrastructure deployment template, which was inadvertently destroyed during a rebase.  

  - Rename of the deployment template to better reflect its purpose.  

  - Moving local assets from "build" into "assets" to better reflect their identity and avoid the root .gitignore pattern for the directory name `[Bb]uild`.  

## Last Upstream Rebase

Wednesday, March 13, 2019  04:43pm(EDT)

## Related and Follow-Up Issues
- [Service Bus Client: Enhance Readme](https://github.com/Azure/azure-sdk-for-net/issues/5349)  (#5349)

### CODE REVIEW

File: src/SDKs/ServiceBus/data-plane/tests/Microsoft.Azure.ServiceBus.Tests/API/APIApprovals.cs [hunk 1]
```diff
@@ -22,9 +27,19 @@ public class ApiApprovals
         #endif
         public void ApproveAzureServiceBus()
         {           
-            var assembly = typeof(Message).Assembly;
+            var assembly = typeof(Message).Assembly;
```
Comment (mikeharder) [inline src/SDKs/ServiceBus/data-plane/tests/Microsoft.Azure.ServiceBus.Tests/API/APIApprovals.cs]: nit: trailing whitespace

File: src/SDKs/ServiceBus/data-plane/tests/Microsoft.Azure.ServiceBus.Tests/API/APIApprovals.cs [hunk 2]
```diff
@@ -22,9 +27,19 @@ public class ApiApprovals
...
-            var assembly = typeof(Message).Assembly;
+            var assembly = typeof(Message).Assembly;      
             var publicApi = Filter(PublicApiGenerator.ApiGenerator.GeneratePublicApi(assembly, whitelistedNamespacePrefixes: new[] { "Microsoft.Azure.ServiceBus." }));  
-            Approvals.Verify(publicApi);
+                                     
+            try
+            {
+              Approvals.Verify(publicApi);
+            }
+
```
Comment (mikeharder) [inline src/SDKs/ServiceBus/data-plane/tests/Microsoft.Azure.ServiceBus.Tests/API/APIApprovals.cs]: Our typical style guidance would be to remove the newline between the `try` and `finally` blocks.  However, if the existing convention in this library is to insert a newline, this is fine for now until we fix all the files to match our style guidelines.

File: src/SDKs/ServiceBus/data-plane/tests/Microsoft.Azure.ServiceBus.Tests/API/APIApprovals.cs [hunk 3]
```diff
@@ -33,5 +48,31 @@ string Filter(string text)
...
+        /// 
+        void CleanApprovalsTempFiles(string approvalsWorkingdDirectory)
+        {
+            foreach (var file in Directory.EnumerateFiles(approvalsWorkingdDirectory, $"{ nameof(ApiApprovals) }.*received.txt", SearchOption.AllDirectories))
+            {
+                try
+                {
+                    File.Delete(file);
+                }
+
```
Comment (mikeharder) [inline src/SDKs/ServiceBus/data-plane/tests/Microsoft.Azure.ServiceBus.Tests/API/APIApprovals.cs]: Remove newline

**Review by mikeharder — APPROVED** [review]
Approve pending whitespace changes

File: src/SDKs/ServiceBus/data-plane/tests/Microsoft.Azure.ServiceBus.Tests/API/APIApprovals.cs (hunk 2 above)
Comment (jsquire) [reply src/SDKs/ServiceBus/data-plane/tests/Microsoft.Azure.ServiceBus.Tests/API/APIApprovals.cs]: That just makes me sad.  Code density hurts readability, IMHO.

Comment (mikeharder) [reply src/SDKs/ServiceBus/data-plane/tests/Microsoft.Azure.ServiceBus.Tests/API/APIApprovals.cs]: I don't know if we have style guidelines specific to the Azure SDK yet, but in the meantime we should follow the .NET Core guidelines:

https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md

File: src/SDKs/ServiceBus/data-plane/tests/Microsoft.Azure.ServiceBus.Tests/API/APIApprovals.cs (hunk 1 above)
Comment (jsquire) [reply src/SDKs/ServiceBus/data-plane/tests/Microsoft.Azure.ServiceBus.Tests/API/APIApprovals.cs]: Nice catch!  Fixed.

File: src/SDKs/ServiceBus/data-plane/tests/Microsoft.Azure.ServiceBus.Tests/API/APIApprovals.cs (hunk 2 above)
Comment (jsquire) [reply src/SDKs/ServiceBus/data-plane/tests/Microsoft.Azure.ServiceBus.Tests/API/APIApprovals.cs]: Fixed.

File: src/SDKs/ServiceBus/data-plane/tests/Microsoft.Azure.ServiceBus.Tests/API/APIApprovals.cs (hunk 3 above)
Comment (jsquire) [reply src/SDKs/ServiceBus/data-plane/tests/Microsoft.Azure.ServiceBus.Tests/API/APIApprovals.cs]: Fixed.

File: src/SDKs/ServiceBus/data-plane/README.md [hunk 4]
```diff
@@ -1,173 +1,52 @@
...
-
-### Issues with Generated Code
-
-Much of the SDK code is generated from metadata specs about the REST APIs. Do not submit PRs that modify generated code. Instead,
-  - File an issue describing the problem,
-  - Refer to the the [AutoRest project](https://github.com/azure/autorest) to view and modify the generator, or
-  - Add additional methods, properties, and overloads to the SDK by adding classes in the 'Customizations' folder of a project
-
-
-![Impressions](https://azure-sdk-impressions.azurewebsites.net/api/impressions/azure-sdk-for-net%2FREADME.png)
```
[the remark this answers is not in the document]
Comment (jsquire) [reply src/SDKs/ServiceBus/data-plane/README.md]: Done.

**Pull request merged**